spec.md

```diff
@@ -230,6 +245,49 @@ A common use of an externally defined module is to represent a Verilog module
 that will be written separately and provided together with FIRRTL-generated
 Verilog to downstream tools.
 
+## Public Modules
+
+A public module is accessible by `Circuit`s other than the `Circuit` it is
```

Comment:
> How does this mesh with the "circuit is a complete design".

Is a circuit a complete design? I don't think that's stated in the spec... Was that the intention in the past? It seems like that notion is no longer true.

> From later: "Modules have the property that instances can always be inlined into the parent module without affecting the semantics of the circuit." How is this valid for separately compiled modules?

What does "the semantics of a circuit" mean? If it means how the circuit is referenced I think that the statement still holds? If it means the behavior of the circuit, then the statement was never true.

> From later: "To disallow infinitely recursive hardware, modules cannot contain instances of itself, either directly, or indirectly through instances of other modules it instantiates." How is this valid for separately compiled modules?

I guess we can't guarantee anything about `extmodules` that are defined in Verilog, since MFC can't check those. I think for `public` modules, do you think it's possible to put in checks to prevent recursive `public` modules?